libraries/rush-lib/src/cli/test/RushCommandLineParserFailureCases.test.ts

```diff
@@ -0,0 +1,67 @@
+// Copyright (c) Microsoft Corporation. All rights reserved. Licensed under the MIT license.
+// See LICENSE in the project root for license information.
+
+// Mock child_process so we can verify tasks are (or are not) invoked as we expect
+jest.mock('child_process');
+
+jest.mock(`@rushstack/package-deps-hash`, () => {
+  return {
+    getRepoRoot(dir: string): string {
+      return dir;
+    },
+    getRepoStateAsync(): ReadonlyMap<string, string> {
+      return new Map();
+    },
+    getRepoChangesAsync(): ReadonlyMap<string, string> {
+      return new Map();
+    }
+  };
+});
+
+import { FileSystem, JsonFile } from '@rushstack/node-core-library';
+import { Autoinstaller } from '../../logic/Autoinstaller';
+import type { ITelemetryData } from '../../logic/Telemetry';
+import type { IParserTestInstance } from './util';
+import { getCommandLineParserInstanceAsync, setSpawnMock } from './util';
+
+describe('RushCommandLineParserFailureCases', () => {
+  describe('execute', () => {
+    afterEach(() => {
+      jest.clearAllMocks();
+    });
+
+    describe('in repo plugin custom flushTelemetry', () => {
+      it('custom telemetry reports errors', async () => {
+        const repoName: string = 'tapFlushTelemetryAndRunBuildActionRepo';
+
+        // WARNING: This test case needs the real implementation of _reportErrorAndSetExitCode.
+        // As a result, process.exit needs to be explicitly mocked to prevent the test runner from exiting.
+        const procProm = new Promise<void>((resolve, reject) => {
+          jest.spyOn(process, 'exit').mockImplementation((() => {
+            resolve();
+            // eslint-disable-next-line @typescript-eslint/no-explicit-any
+          }) as any);
+        });
+
+        const instance: IParserTestInstance = await getCommandLineParserInstanceAsync(repoName, 'build');
+
+        const telemetryFilePath: string = `${instance.parser.rushConfiguration.commonTempFolder}/test-telemetry.json`;
+        FileSystem.deleteFile(telemetryFilePath);
+
+        jest.spyOn(Autoinstaller.prototype, 'prepareAsync').mockImplementation(async function () {});
+        expect(true).toEqual(true);
+
+        setSpawnMock({ emitError: true, returnCode: 1 });
```

Comment:
```suggestion
        setSpawnMock({ returnCode: 1 });
```

I think `emitError` is causing the writing to the process stderr, which shows up as success w/ warnings after everything finishes.